sdk/src/assertions/bmff_hash.rs

```diff
@@ -1208,11 +1230,310 @@ impl BmffHash {
                 _ => return Err(Error::UnsupportedType),
             },
             hashes: VecByteBuf(hashes),
+            fixed_block_size: None,
+            variable_block_sizes: None,
         };
         self.merkle = Some(vec![mm]);
 
         Ok(())
     }
+
+    // create Merkle tree for MerkleMap
+    fn create_merkle_tree_for_merkle_map(
+        &self,
+        reader: &mut dyn CAIRead,
+        box_info: &BoxInfoLite,
+        merkle_map: &mut MerkleMap,
+    ) -> crate::Result<C2PAMerkleTree> {
+        let alg = merkle_map
+            .alg
+            .as_ref()
+            .or(self.alg.as_ref())
+            .ok_or(Error::BadParam("alg is required".to_string()))?;
+
+        // generate leaves in the Merkle tree based on the box length and fixed or variable block sizes
+        let mut leaves = Vec::new();
+        if let Some(fixed_block_size) = merkle_map.fixed_block_size {
```

Comment:
Maybe we should do additional validation somewhere so someone can't accidentally specify both fixed block and variable block size.